frontend/src/components/Collections/index.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { FC } from "react";
+import { useCollections } from "src/common/queries/collections";
+import Collection from "./components/Collection";
+import Heading from "./components/Heading";
+
+const Collections: FC = () => {
+  const { isFetching, data: collections } = useCollections();
+
+  if (isFetching && !collections) return <div>Loading projects...</div>;
```

Comment:
For my understanding: If we are fetching and there aren't any collections, that means that the cache was invalidated b/c of the introduction/removal/edit of a collection?

Reply:
Great question! When `(isFetching && !collections)`, it means it's the first time `react-query` makes an API call, and therefore no cache is available for the view component to render. This pattern is called [`stale-while-revalidate`](https://web.dev/stale-while-revalidate/#:~:text=stale%2Dwhile%2Drevalidate%20helps%20developers,are%20used%20in%20the%20future.&text=Support%20for%20setting%20stale%2Dwhile,Chrome%2075%20and%20Firefox%2068.), which allows the UI to show the cached data while `react-query` is validating and fetching new data in the background

When we mark a `query` to be invalidated, `react-query` will still keep the stale data available for the UI to render, but then it will re-fetch data in the background. If the `query` is still valid, then it won't re-fetch!

Hope the explanation makes sense 😆 If not, please let me know! Thank you!

Reply:
SHOOT I meant to type "there **are** collections"